### ISSUE TITLE
Can't branch from an interhited package

### ISSUE THREAD

**Issue by Vogtinator** [openSUSE/open-build-service#7412]
`openSUSE:Leap:15.1:ARM` inherits from `openSUSE:Leap:15.1`. I want to test a build against the arm project and its repositories though, so I branch from `:ARM` instead.

This does not work:

```
> osc branch --nodevelproject openSUSE:Leap:15.1:ARM go1.12

Note: The branch has been created of a different project,
              openSUSE:Leap:15.1,
      which is the primary location of where development for
      that package takes place.
      That's also where you would normally make changes against.
      A direct branch of the specified package can be forced
      with the --nodevelproject option.
```

osc uses the correct API call:

```
https://api.opensuse.org/source/openSUSE:Leap:15.1:ARM/go1.12?cmd=branch&ignoredevel=1
```

**Comment by adrianschroeter** [comment]
well, the text message of osc is misleading, but the behaviour is correct.

osc detects that it is not same source project.

What you want has nothing to do with devel or incident projects, you want to set the link target directly to :ARM, right?

**Comment by Vogtinator** [comment]
> What you want has nothing to do with devel or incident projects, you want to set the link target directly to :ARM, right?

Yes, and build against `openSUSE:Leap:15.1:ARM`'s repos instead of `openSUSE:Leap:15.1`'s.

**Comment by adrianschroeter** [comment]
you may want to consider to set OBS:BranchTarget attribute in that project to declare it as code stream project. this would solve your issue as well.
